UnityProject/Assets/Tests/Asset/SpellListTest.cs

```diff
@@ -0,0 +1,28 @@
+using System.Collections;
+using System.Collections.Generic;
+using NUnit.Framework;
+using ScriptableObjects.Systems.Spells;
+using Tests;
+using UnityEngine;
+
+public class SpellListTest
+{
+	[Test]
+	public void SpellListHasAllSpells()
+	{
+		var report = new TestReport();
+
+		var AllSpells = Utils.FindAssetsByType<SpellData>();
+
+
+		foreach (var Spell in AllSpells)
+		{
+			if (SpellList.Instance.Spells.Contains(Spell) == false)
```

Comment:
```cs
if (SpellList.Instance.Spells.Contains(Spell)) continue;
report.Fail().AppendLine($"Spell {Spell} is missing From the SpellList Spells SO , Please add in ");

Reply:
It's pretty much the same code it's only a minor difference

Reply:
it's easier to read in the way i should and reduces `{ }`

Reply:
{ } aren't necessarily bad

Reply:
There's also FailIfNot.
```
report.FailIfNot(SpellList.Instance.Spells.Contains(Spell))
	.AppendLine($"Spell {Spell} is missing From the SpellList Spells SO , Please add in ");
```

Reply:
Does that feel straightaway?

Reply:
In this case, it will still iterate through the whole list and append each missing spell. It's functionally similar to the if statement and Fail().AppendLine().